### PR TITLE
redis: Fix types of some options

### PR DESCRIPTION
**Pull request by kajinamit**
The socket_keepalive option in RedisClient is not an integer but a boolean. Also the healthcheck_interval option can accept not only integers but also float values. This fixes the types to parse these options accordingly.

Note this change still keeps support for socket_keepalive=1 or socket_keepalive=0 , but would remove support for the other integer values such as 2 or -1.

### CODE REVIEW

File: gnocchi/common/redis.py [hunk 1]
```diff
@@ -55,19 +55,19 @@
...
 CLIENT_INT_ARGS = frozenset([
     'db',
-    'health_check_interval',
-    'socket_keepalive',
 ])
 
 #: Client arguments that are expected to be float convertible.
 CLIENT_FLOAT_ARGS = frozenset([
     'socket_timeout',
+    'health_check_interval',
```
Comment (rafaelweingartner) [inline gnocchi/common/redis.py:70]: Is this change intended?

Comment (rafaelweingartner) [reply gnocchi/common/redis.py:70]: I see. I missed the second sentence :expressionless:. Sorry for that.

**Review by rafaelweingartner — APPROVED** [review]

**Review by tobias-urdin — APPROVED** [review]

**Comment by tobias-urdin** [comment]
@mergifyio backport stable/4.6
